Port all conda recipes to `rattler-build` & use strict channel priority

gforsyth opened the pull request:
Port all condabuild recipes over to use `rattler-build` instead.

Contributes to rapidsai/build-planning#47

gforsyth committed:
feat(libcucim): port libcucim conda-build to rattler-build

gforsyth committed:
refactor(cucim): update license in cucim

gforsyth committed:
refactor(libcucim): prefix_detection and overlinking checks

gforsyth committed:
refactor(cucim): port from conda-build to rattler-build

gforsyth committed:
ci: switch to `rattler-build`

gforsyth committed:
chore(conda): remove conda-build meta.yaml and scripts

gforsyth commented:
/ok to test

gforsyth committed:
fix(libcucim): comment out changes to LD_LIBRARY_PATH

gforsyth commented:
/ok to test

gforsyth committed:
refactor: add `make` to both recipes

gforsyth commented:
/ok to test

gforsyth committed:
fix(cuda11): add `cudatoolkit` to `host`

gforsyth commented:
/ok to test

gforsyth commented:
/ok to test

gforsyth committed:
Merge branch 'branch-25.06' into rattler-build

gforsyth committed:
refactor: use strict channel priority

jakirkham commented:
We discussed this again at the cuCIM meeting this week, but didn't see an obvious cause for the test failures

Diffing the changes relative to a recent passing PR ( https://github.com/rapidsai/cucim/pull/867 ) didn't provide any obvious clues

Tried updating this PR and opening a fresh testing PR ( https://github.com/rapidsai/cucim/pull/868 ). The latter passed whereas this still failed

Am starting wondering if there is something unique to the changes in this PR, which lead to the CI failures we see. Still not sure exactly what change causes it though

gforsyth commented:
> We discussed this again at the cuCIM meeting this week, but didn't see an obvious cause for the test failures
> 
> Diffing the changes relative to a recent passing PR ( #867 ) didn't provide any obvious clues
> 
> Tried updating this PR and opening a fresh testing PR ( #868 ). The latter passed whereas this still failed
> 
> Am starting wondering if there is something unique to the changes in this PR, which lead to the CI failures we see. Still not sure exactly what change causes it though

Ok, thanks for looking into that @jakirkham !  I'll start combing through the recipe with a fine-toothed comb

gforsyth committed:
chore(libcucim): add binutils to build

gforsyth commented:
/ok to test

gforsyth committed:
Merge branch 'branch-25.08' into rattler-build

jakirkham reviewed (commented):
Thanks Gil! 🙏

Noticed something different about how the S3 environment variables are set. Could you please take a closer look?

jakirkham commented on conda/recipes/cucim/meta.yaml:
```diff
@@ -1,100 +0,0 @@
-    - CMAKE_CUDA_COMPILER_LAUNCHER
-    - CMAKE_CXX_COMPILER_LAUNCHER
-    - CMAKE_GENERATOR
-    - PARALLEL_LEVEL
```
Also should we still set `PARALLEL_LEVEL`?

Looks like it was dropped from the new recipe

jakirkham commented:
When you have a moment Gil, could you please resolve the conflicts here?

gforsyth committed:
Merge branch 'branch-25.08' into rattler-build

gforsyth committed:
chore: update `libcucim` rattler recipe

gforsyth committed:
chore: update `cucim` rattler recipe

gforsyth committed:
chore: remove conda-build recipe and scripts

gforsyth committed:
fix: misplaced quotation mark

gforsyth commented:
/ok to test

jakirkham committed:
Merge branch 'branch-25.10' into rattler-build

jakirkham commented:
/ok to test

jameslamb commented:
Adding a note that #923 should be reverted on this branch whenever this PR is revived.

jakirkham committed:
Merge rapidsai/branch-25.10 into gforsyth/rattler-build

jakirkham committed:
Merge rapidsai/branch-25.10 into gforsyth/rattler-build

jakirkham commented:
/ok to test

jakirkham committed:
Merge rapidsai/branch-25.10 into gforsyth/rattler-build

jakirkham committed:
Merge rapidsai/main into gforsyth/rattler-build

jakirkham committed:
Merge branch 'main' into rattler-build

jakirkham commented:
/ok to test 15422e9c3af510a4af314da2b2dc06a5e21a7c1e

jakirkham committed:
Drop unneeded `CPP_CHANNEL`

jakirkham commented:
/ok to test 71802ee171816e98d9821f2e18e4426de198e5fd

jakirkham committed:
Drop leftover copy-pasta

jakirkham commented:
/ok to test ef5a9fb2014dc64e2048911923425de51a80997f

jakirkham committed:
Fix nvimgcodec version constraint syntax

The variable already contains a range with comparison operators. So
adding `=`s in front just causes issues. Hence the `=`s is dropped.

jakirkham commented:
/ok to test dec1858ad6c24aca790a769da727b066f9654c59

jakirkham committed:
Set RAPIDS package version in Python build

jakirkham commented:
/ok to test d8ed0943f50fc886e039caa2b7546e6d94786a5e

jakirkham committed:
Enable experimental for rattler-build

This is needed for Jinja like `${{ git.head_rev(".")[:8] }}`, which
RAPIDS frequently uses in its package build strings.

jakirkham commented:
/ok to test 4fd8c8f0e33b4910e052b1ec9739205bc4e073f3

jakirkham committed:
Drop extra `--experimental` from `rattler-build`

Looks like this is already being passed in via the rattler environment
variable arguments. So drop the extra manual addition from the C++
builds.

jakirkham commented:
/ok to test 98a044fc6a7b454d62dbc36a397c095eeac267af

jakirkham committed:
Readd channels for rattler-build & cached C++ pkgs

jakirkham commented:
/ok to test 1c78cd6ed892dcbb2966660f3b4f1572469f1e44

jakirkham committed:
Set `RAPIDS_PACKAGE_VERSION` and use it

Make sure `RAPIDS_PACKAGE_VERSION` is always set first and then used for
all later steps. This ensures it is available for the recipe to check
later as well.

jakirkham committed:
Readd sccache bits for C++ build

jakirkham committed:
Write to VERSION file in C++ build

jakirkham committed:
Readd xorg workaround for clobbering (warn)

jakirkham committed:
Reconsolidate `sccache` steps in Python build

jakirkham committed:
Log before building `libcucim`

jakirkham commented:
/ok to test 5313a032404e9fc3ed95ac46b8f5c0e703ee5cbc

jakirkham committed:
Consolidate rattler-build channel config

jakirkham commented:
/ok to test 5448cf76cbdd4f57273ae44e9ab4709d3b389130
